### PR TITLE
Create py.typed to mark this library as typed

### PR DESCRIPTION
**Pull request by ottojo**
This marker file informs type checkers of downstream libraries that launch_ros is typed.
In my use case, this enables using the `strict` typechecking mode in pyright for a downstream project using launch_ros

This was done for rclpy in https://github.com/ros2/rclpy/pull/946

### CODE REVIEW

**Comment by ottojo** [comment]
I'm not quite sure how that caused the failed test in the Rpr__launch_ros__ubuntu_jammy_amd64 CI, that might be unrelated?

Apropos CI: In the linked rclpy PR there is some discussion on testing. Pyright can test libraries for type correctness, but this test currently fails for launch_ros:
```
pyright --verifytypes launch_ros --ignoreexternal

Symbols exported by "launch_ros": 265
  With known type: 75
  With ambiguous type: 3
  With unknown type: 187
    (Ignoring unknown types imported from other packages)
  Functions without docstring: 13
  Functions without default param: 0
  Classes without docstring: 0

Other symbols referenced but not exported by "launch_ros": 0
  With known type: 0
  With ambiguous type: 0
  With unknown type: 0

Type completeness score: 28.3%
```

 It does however also fail for rclpy:
```
pyright --verifytypes rclpy --ignoreexternal

[...]

Symbols exported by "rclpy": 844
  With known type: 233
  With ambiguous type: 13
  With unknown type: 598
    (Ignoring unknown types imported from other packages)
  Functions without docstring: 119
  Functions without default param: 0
  Classes without docstring: 27

Other symbols referenced but not exported by "rclpy": 19
  With known type: 6
  With ambiguous type: 0
  With unknown type: 13

Type completeness score: 27.6%
```

I think it still makes sense to mark this as "supposed to by typed" (if this is the intention, i assume this based on the large amount of type hints), it works well for rclpy.

**Comment by clalancette** [comment]
> I'm not quite sure how that caused the failed test in the Rpr__launch_ros__ubuntu_jammy_amd64 CI, that might be unrelated?

Yeah, it's unrelated.  It should be fixed by #377

File: launch_ros/launch_ros/py.typed [hunk 1]
```diff
@@ -0,0 +1 @@
+
```
Comment (clalancette) [inline launch_ros/launch_ros/py.typed:1]: The `py.typed` file in `rclpy` is completely empty, I would just prefer if we were consistent here.  So I'd say let's remove the carriage-return that is in here.

File: launch_ros/setup.py [hunk 2]
```diff
@@ -12,6 +12,7 @@
         ('share/ament_index/resource_index/packages',
             ['resource/' + package_name]),
     ],
+    package_data={"launch_ros": ["py.typed"]},
```
[the remark this answers is not in the document]
Comment (ottojo) [reply launch_ros/setup.py:15]: It seems to get installed for rclpy, perhaps implicitly by some other method... I will try if installation works without this line for launch_ros

 `-rw-r--r-- 1 root root 0 May 11 13:58 /opt/ros/iron/lib/python3.10/site-packages/rclpy/py.typed`

[the remark this answers is not in the document]
Comment (clalancette) [reply launch_ros/setup.py:15]: Oh!  Sorry I missed that.  Yeah, in that case I'm happy to have it installed here.  I'll resolve this comment.

File: launch_ros/launch_ros/py.typed (hunk 1 above)
Comment (ottojo) [reply launch_ros/launch_ros/py.typed]: done

File: launch_ros/setup.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (ottojo) [reply launch_ros/setup.py:15]: well when comparing with rclpy the first realization is that rclpy is installed by cmake, and while i don't feel like digging through that, it's enough of an explanation for me as to why it needs to be installed explicitly here :smile:

Comment (clalancette) [inline launch_ros/setup.py:15]: flake8 wants single instead of double quotes:

```suggestion
    package_data={'launch_ros': ['py.typed']},
```

**Review by clalancette — CHANGES_REQUESTED** [review]

**Comment by clalancette** [comment]
New CI:
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/19414/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/13942/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/20136/)

**Review by clalancette — APPROVED** [review]

**Comment by clalancette** [comment]
Looks good, thanks for iterating!

**Pull request merged**